src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -748,3 +751,86 @@ impl RustcTargetData {
         self.target_config(kind).links_overrides.get(lib_name)
     }
 }
+
+/// Structure used to deal with Rustdoc fingerprinting
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RustDocFingerprint {
+    pub rustc_vv: String,
+}
+
+impl RustDocFingerprint {
+    /// Read the `RustDocFingerprint` info from the fingerprint file.
+    fn read<'a, 'cfg>(cx: &Context<'a, 'cfg>) -> CargoResult<Self> {
+        let rustdoc_data = paths::read(&cx.files().host_root().join(".rustdoc_fingerprint.json"))?;
+        serde_json::from_str(&rustdoc_data).map_err(|e| anyhow::anyhow!("{:?}", e))
+    }
+
+    /// Write the `RustDocFingerprint` info into the fingerprint file.
+    fn write<'a, 'cfg>(&self, cx: &Context<'a, 'cfg>) -> CargoResult<()> {
+        paths::write(
+            &cx.files().host_root().join(".rustdoc_fingerprint.json"),
+            serde_json::to_string(&self)?.as_bytes(),
+        )
+    }
+
+    fn remove_doc_dirs(doc_dirs: &Vec<PathBuf>) -> CargoResult<()> {
+        let errs: Vec<CargoResult<()>> = doc_dirs
+            .iter()
+            .map(|path| paths::remove_dir_all(&path))
+            .filter(|res| res.is_err())
+            .collect();
+        if !errs.is_empty() {
+            Err(anyhow::anyhow!("Dir removal error"))
+        } else {
+            Ok(())
+        }
```

Comment:
This shouldn't erase the error.  I think it would be fine to just iterate and return the first error encountered. 

Also, this doesn't handle the scenario where a doc dir doesn't exist.

Reply:
True. I don;t know why I overrided the error.. Coding late is not agood idea x)

As for the doc dir that doesn't exist, added `.filter(|path| path.exists())`. This should do the job filtering the paths that would deal into the scenario you defined.